formula-android/src/main/java/com/instacart/formula/android/internal/ActivityManager.kt

```diff
@@ -0,0 +1,111 @@
+package com.instacart.formula.android.internal
+
+import androidx.fragment.app.FragmentActivity
+import androidx.lifecycle.Lifecycle
+import com.instacart.formula.activity.ActivityResult
+import com.instacart.formula.fragment.FragmentEnvironment
+import com.instacart.formula.integration.ActivityStore
+import com.instacart.formula.integration.FragmentFlowRenderView
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.disposables.Disposable
+
+/**
+ * Activity manager connects [ActivityStore] and the [Activity].
+ */
+internal class ActivityManager<Activity : FragmentActivity>(
+    private val environment: FragmentEnvironment,
+    private val delegate: ActivityStoreContextImpl<Activity>,
+    private val store: ActivityStore<Activity>
+) {
+
+    private val fragmentState = store
+        .fragments
+        .state(environment)
+        .doOnNext(delegate.fragmentFlowStateRelay::accept)
+        .replay(1)
+
+    internal val stateSubscription: Disposable
+    private var uiSubscription: Disposable? = null
+    private var fragmentRenderView: FragmentFlowRenderView? = null
+
+    init {
+        stateSubscription = if (store.streams != null) {
+            val disposables = CompositeDisposable()
+            disposables.add(fragmentState.connect())
+            disposables.add(store.streams.invoke())
+            disposables
+        } else {
+            fragmentState.connect()
+        }
+    }
+
+    fun onPreCreate(activity: Activity) {
+        // Give store a chance to initialize the activity.
+        store.configureActivity?.invoke(activity)
+
+        // Initialize render view
+        fragmentRenderView = FragmentFlowRenderView(
+            activity = activity,
+            onLifecycleEvent = {
+                store.fragments.onLifecycleEffect(it)
+                store.onFragmentLifecycleEvent?.invoke(it)
```

Comment:
Interesting, I was thinking the opposite, that we want fragment flow state to be fully updated before we emit the event. One note is that `Removed` event has the last render model, so it might be enough for your use-case.